File: BackEnd/Ipsos/Entities/Parametros/FiltroPadrao.cs
using Entities.Filtros;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entities.Parametros
{
    public class FiltroPadrao
    {
        public List<PadraoComboFiltro> Target { get; set; }

        public List<PadraoComboFiltro> Regiao { get; set; }

        public List<PadraoComboFiltro> Demografico { get; set; }

        public List<PadraoComboFiltro> Onda { get; set; }

        public List<PadraoComboFiltro> Marca { get; set; }

        public int CodUser { get; set; }

        public int CodIdioma { get; set; }

        public int Sequencia { get; set; }

        public int Atributo { get; set; }

        public string TituloGrafico { get; set; }

        public int ParamTipo { get; set; }

        public int ParamOndaAtual { get; set; }

        public int ParamSTB { get; set; }
        public int ParamBIA { get; set; }

        public int ParamDenominators { get; set; }

    }

    public class FiltroPadraoExcel
    {
        public List<PadraoComboFiltro> Target { get; set; }
        public List<PadraoComboFiltro> Regiao { get; set; }
        public List<PadraoComboFiltro> Demografico { get; set; }
        public List<PadraoComboFiltro> Onda { get; set; }
        public List<PadraoComboFiltro> Marca { get; set; }

        public List<PadraoComboFiltro> Onda1 { get; set; }
        public List<PadraoComboFiltro> Onda2 { get; set; }
        public List<PadraoComboFiltro> Onda3 { get; set; }
        public List<PadraoComboFiltro> Onda4 { get; set; }

        public List<PadraoComboFiltro> Marca1 { get; set; }
        public List<PadraoComboFiltro> Marca2 { get; set; }
        public List<PadraoComboFiltro> Marca3 { get; set; }
        public List<PadraoComboFiltro> Marca4 { get; set; }
        public List<PadraoComboFiltro> Marca5 { get; set; }

        public List<PadraoComboFiltro> Marca6 { get; set; }
        public List<PadraoComboFiltro> Marca7 { get; set; }
        public List<PadraoComboFiltro> Marca8 { get; set; }

        public int CodUser { get; set; }

        public int CodIdioma { get; set; }

        public string TituloGrafico { get; set; }

        public int ParamTipo { get; set; }

        public int ParamOndaAtual { get; set; }

        public int ParamBIA { get; set; }

        public int ParamDenominators { get; set; }
    }


    public class FiltroPadraoExcelDuplo
    {
        public List<PadraoComboFiltro> Target { get; set; }
        public List<PadraoComboFiltro> Regiao { get; set; }
        public List<PadraoComboFiltro> Demografico { get; set; }
        public List<PadraoComboFiltro> Onda { get; set; }
        public List<PadraoComboFiltro> Marca { get; set; }

        public PadraoComboFiltro OndaDuploColuna1 { get; set; }
        public PadraoComboFiltro OndaDuploColuna1_2 { get; set; }
        public PadraoComboFiltro OndaDuploColuna1_3 { get; set; }

        public PadraoComboFiltro OndaDuploColuna2 { get; set; }
        public PadraoComboFiltro OndaDuploColuna2_2 { get; set; }
        public PadraoComboFiltro OndaDuploColuna2_3 { get; set; }

        public PadraoComboFiltro OndaDuploColuna3 { get; set; }
        public PadraoComboFiltro OndaDuploColuna3_2 { get; set; }
        public PadraoComboFiltro OndaDuploColuna3_3 { get; set; }

        public PadraoComboFiltro OndaDuploColuna4 { get; set; }
        public PadraoComboFiltro OndaDuploColuna4_2 { get; set; }
        public PadraoComboFiltro OndaDuploColuna4_3 { get; set; }

        public PadraoComboFiltro OndaDuploColuna5 { get; set; }
        public PadraoComboFiltro OndaDuploColuna5_2 { get; set; }
        public PadraoComboFiltro OndaDuploColuna5_3 { get; set; }

        public PadraoComboFiltro MarcaDuploColuna1 { get; set; }
        public PadraoComboFiltro MarcaDuploColuna2 { get; set; }
        public PadraoComboFiltro MarcaDuploColuna3 { get; set; }
        public PadraoComboFiltro MarcaDuploColuna4 { get; set; }
        public PadraoComboFiltro MarcaDuploColuna5 { get; set; }

        public int CodUser { get; set; }

        public int CodIdioma { get; set; }

        public string TituloGrafico { get; set; }
    }
    public class FiltroPadraoFullLoad
    {
        public FiltroPadrao FiltroPadraoColuna1 { get; set; }
        public FiltroPadrao FiltroPadraoColuna2 { get; set; }
        public FiltroPadrao FiltroPadraoColuna3 { get; set; }
        public FiltroPadrao FiltroPadraoColuna4 { get; set; }
        public FiltroPadrao FiltroPadraoColuna5 { get; set; }
    }

}

File: FrontEnd/src/app/models/Filtros/FiltroPadrao.ts
import { PadraoComboFiltro } from "../padrao-combo-filtro/padrao-combo-filtro";

export class FiltroPadrao {

    constructor() {
        this.Target = new Array<PadraoComboFiltro>();
        this.Regiao = new Array<PadraoComboFiltro>();
        this.Demografico = new Array<PadraoComboFiltro>();
        this.Onda = new Array<PadraoComboFiltro>();
        this.Marca = new Array<PadraoComboFiltro>();
    }

    Target: Array<PadraoComboFiltro>;
    Regiao: Array<PadraoComboFiltro>;
    Demografico: Array<PadraoComboFiltro>;
    Onda: Array<PadraoComboFiltro>;
    Marca: Array<PadraoComboFiltro>;
    CodUser: number;
    CodIdioma: number;
    Sequencia: number;
    Atributo: number;
    TituloGrafico: string;
    ParamTipo:number;
    ParamOndaAtual :number;
    ParamSTB:number;
    ParamBIA:number;
    ParamDenominators:number;
}


export class FiltroPadraoExcel {

    constructor() {
        this.Target = new Array<PadraoComboFiltro>();
        this.Regiao = new Array<PadraoComboFiltro>();
        this.Demografico = new Array<PadraoComboFiltro>();
        this.Onda = new Array<PadraoComboFiltro>();
        this.Marca = new Array<PadraoComboFiltro>();

        this.Onda1 = new Array<PadraoComboFiltro>();
        this.Onda2 = new Array<PadraoComboFiltro>();
        this.Onda3 = new Array<PadraoComboFiltro>();
        this.Onda4 = new Array<PadraoComboFiltro>();

        this.Marca1 = new Array<PadraoComboFiltro>();
        this.Marca2 = new Array<PadraoComboFiltro>();
        this.Marca3 = new Array<PadraoComboFiltro>();
        this.Marca4 = new Array<PadraoComboFiltro>();
        this.Marca5 = new Array<PadraoComboFiltro>();

        this.Marca6 = new Array<PadraoComboFiltro>();
        this.Marca7 = new Array<PadraoComboFiltro>();
        this.Marca8 = new Array<PadraoComboFiltro>();

        
    }

    Target: Array<PadraoComboFiltro>;
    Regiao: Array<PadraoComboFiltro>;
    Demografico: Array<PadraoComboFiltro>;
    Onda: Array<PadraoComboFiltro>;
    Marca: Array<PadraoComboFiltro>;

    Onda1: Array<PadraoComboFiltro>;
    Onda2: Array<PadraoComboFiltro>;
    Onda3: Array<PadraoComboFiltro>;
    Onda4: Array<PadraoComboFiltro>;

    Marca1: Array<PadraoComboFiltro>;
    Marca2: Array<PadraoComboFiltro>;
    Marca3: Array<PadraoComboFiltro>;
    Marca4: Array<PadraoComboFiltro>;
    Marca5: Array<PadraoComboFiltro>;

    Marca6: Array<PadraoComboFiltro>;
    Marca7: Array<PadraoComboFiltro>;
    Marca8: Array<PadraoComboFiltro>;

    CodUser: number;
    CodIdioma: number;

    TituloGrafico: string;

    ParamTipo:number;
    ParamOndaAtual:number;
    ParamBIA :number;

    ParamDenominators:number;
}

export class FiltroPadraoFullLoad {
    FiltroPadraoColuna1: FiltroPadrao;
    FiltroPadraoColuna2: FiltroPadrao;
    FiltroPadraoColuna3: FiltroPadrao;
    FiltroPadraoColuna4: FiltroPadrao;
    FiltroPadraoColuna5: FiltroPadrao;
}


export class FiltroPadraoExcelDuplo {

    constructor() {
        this.Target = new Array<PadraoComboFiltro>();
        this.Regiao = new Array<PadraoComboFiltro>();
        this.Demografico = new Array<PadraoComboFiltro>();
        this.Onda = new Array<PadraoComboFiltro>();
        this.Marca = new Array<PadraoComboFiltro>();

        this.OndaDuploColuna1 = new PadraoComboFiltro();
        this.OndaDuploColuna1_2 = new PadraoComboFiltro();

        this.OndaDuploColuna2 = new PadraoComboFiltro();
        this.OndaDuploColuna2_2 = new PadraoComboFiltro();

        this.OndaDuploColuna3 = new PadraoComboFiltro();
        this.OndaDuploColuna3_2 = new PadraoComboFiltro();

        this.OndaDuploColuna4 = new PadraoComboFiltro();
        this.OndaDuploColuna4_2 = new PadraoComboFiltro();

        this.OndaDuploColuna5 = new PadraoComboFiltro();
        this.OndaDuploColuna5_2 = new PadraoComboFiltro();

        this.MarcaDuploColuna1 = new PadraoComboFiltro();
        this.MarcaDuploColuna2 = new PadraoComboFiltro();
        this.MarcaDuploColuna3 = new PadraoComboFiltro();
        this.MarcaDuploColuna4 = new PadraoComboFiltro();
        this.MarcaDuploColuna5 = new PadraoComboFiltro();
    }

    Target: Array<PadraoComboFiltro>;
    Regiao: Array<PadraoComboFiltro>;
    Demografico: Array<PadraoComboFiltro>;
    Onda: Array<PadraoComboFiltro>;
    Marca: Array<PadraoComboFiltro>;

    // Filtros de Marca para utilização da geração do Excel Grafico Comparativo Marcas 
    OndaDuploColuna1: PadraoComboFiltro;
    OndaDuploColuna1_2: PadraoComboFiltro;
    OndaDuploColuna1_3: PadraoComboFiltro;

    OndaDuploColuna2: PadraoComboFiltro;
    OndaDuploColuna2_2: PadraoComboFiltro;
    OndaDuploColuna2_3: PadraoComboFiltro;

    OndaDuploColuna3: PadraoComboFiltro;
    OndaDuploColuna3_2: PadraoComboFiltro;
    OndaDuploColuna3_3: PadraoComboFiltro;

    OndaDuploColuna4: PadraoComboFiltro;
    OndaDuploColuna4_2: PadraoComboFiltro;
    OndaDuploColuna4_3: PadraoComboFiltro;

    OndaDuploColuna5: PadraoComboFiltro;
    OndaDuploColuna5_2: PadraoComboFiltro;
    OndaDuploColuna5_3: PadraoComboFiltro;

    // Filtros de Marca para utilização da geração do Excel Grafico Comparativo Marcas 
    MarcaDuploColuna1: PadraoComboFiltro;
    MarcaDuploColuna2: PadraoComboFiltro;
    MarcaDuploColuna3: PadraoComboFiltro;
    MarcaDuploColuna4: PadraoComboFiltro;
    MarcaDuploColuna5: PadraoComboFiltro;


    CodUser: number;
    CodIdioma: number;

    TituloGrafico: string;
}

